src/ray/core_worker/task_manager.cc

```diff
@@ -640,6 +659,16 @@ void TaskManager::MarkDependenciesResolved(const TaskID &task_id) {
   }
 }
 
+void TaskManager::MarkTaskSubmitted(const TaskID &task_id) {
+  absl::MutexLock lock(&mu_);
+  auto it = submissible_tasks_.find(task_id);
+  RAY_CHECK(it != submissible_tasks_.end())
```

Comment:
hmm RAY_CHECK seems a bit extreme. Could we return bool and return early instead?